Sources/Core/OAFavoritesMapLayerProvider.mm

```diff
@@ -68,7 +68,16 @@
     QString iconName = isFullSize ? fav->getIcon() : QStringLiteral("");
     QString backgroundIconName = backgroundImageNameByType(fav->getBackground());
     QString size = isFullSize ? QStringLiteral("_full") : QStringLiteral("_small");
-    QString iconId = QString::number(fav->getColor().a + fav->getColor().r + fav->getColor().g + fav->getColor().b) + QStringLiteral("_") + iconName + QStringLiteral("_") + backgroundIconName + size + QString("_%1").arg(_symbolsScaleFactor, 0, 'f', 2);
+    
+    auto color = fav->getColor();
+    if (color.argb == 0)
+    {
+        CGFloat r,g,b,a;
+        [[OADefaultFavorite getDefaultColor] getRed:&r green:&g blue:&b alpha:&a];
+        color = OsmAnd::FColorARGB(a,r,g,b);
+    }
+    
+    QString iconId = QString::number(color.a + color.r + color.g + color.b) + QStringLiteral("_") + iconName + QStringLiteral("_") + backgroundIconName + size + QString("_%1").arg(_symbolsScaleFactor, 0, 'f', 2);
```

Comment:
Better to use `arg` for each parameter and format color like %1_%2_%3_%4 (instead of addition)